File: CheckPermutation.java
/**
 * Check if two strings are permutations of one another
 */

import java.util.*;
import java.lang.*;

public class CheckPermutation {

    public static void main(String[] args) {
        System.out.println(isCheckPermReturningTrueWhenStringsArePermutations());
        System.out.println(isCheckPermReturningFalseWhenStringsAreNotPermutations());
    }

    static boolean checkPermutation(String str1, String str2) {
        if (str1.length() != str2.length()) {
            return false;
        }

        char[] firstStr = str1.toCharArray();
        char[] secondStr = str2.toCharArray();

        Arrays.sort(firstStr);
        Arrays.sort(secondStr);

        for (int i = 0; i < str1.length(); i++) {
            if (firstStr[i] != secondStr[i]) {
                return false;
            }
        }
        
        return true;
    }

    
    /* --- --- TESTS --- --- */

    static boolean isCheckPermReturningTrueWhenStringsArePermutations() {
        if(checkPermutation("barrel", "rerlab") == true) {
            return true;
        }
        
        return false;
    }

    static boolean isCheckPermReturningFalseWhenStringsAreNotPermutations() {
        if(checkPermutation("racecar", "orange") == false) {
            return true;
        }
        
        return false;
    }
}

File: checkPermutation.py
"""Check if two strings are permutations of one another
"""

def preprocess(string):
    return sorted(string.lower())

def check_permutation(string1, string2):
    s1 = ''.join(preprocess(string1))
    s2 = ''.join(preprocess(string2))

    if s1 == s2:
        return True
    else:
        return False

"""TESTS"""

def returns_false_if_not_permutations():
    if check_permutation("door", "racecar") == False:
        return 1
    else:
        return 0

def returns_true_if_permutations():
    if check_permutation("door", "odor") == True:
        return 1
    else:
        return 0
    
def test_check_permutation():
    print(returns_false_if_not_permutations())
    print(returns_true_if_permutations())

test_check_permutation()